src/4D_api.cpp

```diff
@@ -1448,11 +1448,18 @@ int proj_errno_reset (const PJ *P) {
 }
 
 
-/* Create a new context */
+/* Create a new context based on the default context */
 PJ_CONTEXT *proj_context_create (void) {
     return pj_ctx_alloc ();
 }
 
+/* Create a new context based on a custom context */
+PJ_CONTEXT *proj_context_clone (PJ_CONTEXT *ctx) {
+    if (nullptr==ctx)
+        return pj_ctx_alloc ();
+
+    return pj_ctx_clone (ctx);
```

Comment:
this function should be moved to ctx.cpp() and merged with it

Reply:
:+1: